### DISCUSSION TITLE
empty out.xml

### DISCUSSION THREAD

**Discussion by TAkbay** [DamianOsipiuk/testcafe-reporter-testrail#32] [Q&A] [answered]
Hi!

I'm new to Testcafe and Testrail and I really want to use this plugin to have a smooth automation process, but I don't get any updates in Testrail :(
When I execute `yarn testcafe --config-file .testrailrc.json -r spec,testrail:out.xml chrome .\Testcafe\Testrail\Login.js` the test starts and the out.xml file is getting created, but it's completely empty and I don't understand why.

Following versions do I use:
In my testcafe project 
    "testcafe": "1.18.3",
    "testcafe-reporter-testrail": "^0.9.0"
Testrail version: v7.4.0.7005

My config file .testrailrc.json looks like this:
```json
{
    "enabled": true,
    "host":"https://URL.com/",
    "user": "EMAIL",
    "apiKey":"KEY",
    "projectId":"P5",
    "runId": "R47",
    "caseMeta": "CID",
    "updateRunTestCases": true
}
```
My test Login.js:

```js
import { Selector } from 'testcafe'

fixture`Login`.page`https://URL.com`

test.meta({ CID: 'C1006' })('Login', async (t) => {
  const usernameInput = Selector('#username')
  await t.typeText(usernameInput, 'thisIsMyUsername')
  await t.expect(usernameInput.value).eql('thisIsMyUsername', 'input is filled')
})
```

Did I forgot something or am I doing something wrong?

Thanks in advance!

**Comment by DamianOsipiuk** [comment]
Hello @TAkbay

I think you should look at your Testrail instance and check if test results are added to your test run.
This plugin connects directly via Testrail API. The file that you are specifying `out.xml` is only a workaround due to a limitation of testcafe: 

https://testcafe.io/documentation/402638/reference/configuration-file#reporter

```
You can use multiple reporters, but note that only one reporter can write to stdout. All other reporters must output to files.
```

Since you still want the output to console via `spec` reporter, you have to cheat testcafe and tell that this plugin would output to file, while in fact it's not.

Additionally, you do not have to specify the `--config-file .testrailrc.json` flag, as it's for the testcafe config, not for this plugin and it might break something.

If that still does not work, try setting `updateRunTestCases` to `false` and check whether new test run gets created in your testrail project.

**Reply by TAkbay** [reply]
Hi @DamianOsipiuk,
thanks for your reply! I tried your suggestions but it didn't work :( My test runs are still not getting updated. I enabled the API in the site settings of Testrail too, but I will take a closer look what could "block" the incoming informations.
The strange thing is, that I tried to speak with testrail via REST calls with SOAPUI and it worked.. :/ 
If I find the issue, I will share it here.

**Reply by TAkbay** [reply]
This time I tried to do a simple GET request with xhr2 to check if any other code would work. So I tried the following code and I'm getting the status 200 with it.

```javascript
const XMLHttpRequest = require('xhr2')
const xhr = new XMLHttpRequest()
const url = 'https://<TESTRAIL_INSTANCE>/index.php?/api/v2/get_case/1006'
xhr.open('GET', url)
xhr.setRequestHeader(
  'Authorization',
  'Basic ' + Buffer.from('username:password').toString('base64')
)
xhr.send()

xhr.onreadystatechange = (e) => {
  console.log(xhr.status)
  console.log(xhr.responseText)
}
```
So there must be anything with this plugin. But what :/

**Reply by DamianOsipiuk** [reply]
Have you tried to set `updateRunTestCases` to `false` in your config?

**Reply by TAkbay** [reply]
Yes :(

**Reply by DamianOsipiuk** [reply]
Some things that might help and are kinda `better check this twice even if you did it previously` :)
- do you see any console errors prefixed with `[TestRail]` during test run?
- try to remove trailing slash from `host`: `"host":"https://URL.com/",` -> `"host":"https://URL.com",` 
- generate new `api key` for your user and put it into the config
- make sure that `api access` is enabled for your user
- make sure that you have enabled ability to `edit test results` in site settings

**Comment by TAkbay** [comment, marked as answer]
Found the solution I just checked your code how the plugin is working.
I had a .testrailrc.json but it should be called .testrailrc
So I misunterstood something.
The second thing was, that I needed to provide a suite ID. So I changed the project configuration in testrail and used the suite which was created by default.
